### PR TITLE
Fixed incorrect truncated tlv error that prevents reading empty (0 byte) TLVs

### PR DESCRIPTION
**Pull request by astromechza**
We hit this issue when reading a 0 byte custom TLV value. Testing identified that this was because the length was 0 bytes and the length check in this PR was affecting it.

### CODE REVIEW

**Comment by emersion** [comment]
This change looks good. Can you add a test with a 0-length TLV?

**Comment by astromechza** [comment]
@emersion I've added the test now and checked that it fails on the old code 👍 

Any idea how we get a new release cut?

**Comment by astromechza** [comment]
@pires what do you think of this diff?

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Comment by emersion** [comment]
>Any idea how we get a new release cut?

Once this PR is merged, you can just `go get github.com/pires/go-proxyproto@<commit>` in your project to get the fix. Or wait for @pires to tag a new release.

**Review by pires — APPROVED** [review]
This is a great catch, thanks a lot, @astromechza!

**Comment by pires** [comment]
Tagged v0.3.3 and will handle release notes over the coming days.

**Pull request merged**
